File: code-examples/examples/example-2/src/main/java/be/swsb/typesafety/example2/domain/booking/BookingRepository.java
package be.swsb.typesafety.example2.domain.booking;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class BookingRepository {

    private List<Booking> bookings = new ArrayList<>();

    public UUID save(final Booking booking) {
        //booking is saved to a database or whatever
        bookings.add(booking);
        return booking.getId();
    }

    public List<Booking> findUnsentBookings() {
        return bookings;
    }
}

File: code-examples/examples/example-5/src/main/java/be/swsb/typesafety/example5/domain/booking/BookingRepository.kt
package be.swsb.typesafety.example5.domain.booking

import java.util.*

open class BookingRepository {

    private val bookings = ArrayList<Booking>()

    open fun save(booking: Booking): UUID {
        bookings.add(booking)
        return booking.id
    }

    open fun findUnsentBookings(): List<Booking> {
        return bookings
    }
}
